Source/DafnyDriver/Commands/FilterOption.cs

```diff
@@ -0,0 +1,24 @@
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Boogie;
+
+namespace Microsoft.Dafny; 
+
+public class FilterOption : CommandLineOption<IEnumerable<string>> {
+  public static readonly FilterOption Instance = new();
+
+  public override object DefaultValue => Enumerable.Empty<string>();
+  public override string LongName => "filter";
+  public override string ArgumentName => "pattern";
+
+  public override string Description => @"
+(experimental) Only check proofs matched by pattern <p>. This option may be specified multiple times to match multiple patterns. The pattern <p> may contain * wildcards which match any character zero or more times.".TrimStart();
+  public override void Parse(CommandLineParseState ps, DafnyOptions options) {
+    throw new System.NotImplementedException();
+  }
+
+  public override string PostProcess(DafnyOptions options) {
+    options.ProcsToCheck.AddRange(Get(options).Select(s => "*" + s + "*"));
```

Comment:
Do we really want to always add `*`?

Reply:
I'm not sure why the * at the end is necessary, but I can't get anything to pass the filter without it. The * at the start is because Dafny adds internally defined prefixes to the proof names when translating them to Boogie, so I don't think we can do without it.

Reply:
I think ideally, the configured pattern is used to do a symbol search at the Dafny level, and then Dafny's Boogie translation machinery is invoked to get the Boogie name for this symbol, which is passed to Boogie's `ProcsToCheck`. But building that doesn't seem like it's worth the effort run now, since this works well enough.

Reply:
But it's convenient to be able to match on specific prefixes, so always adding the `*` isn't ideal; could we just recommend adding them in the doc?

Reply:
Are you suggesting to remove the automatic prefix and postfix wildcards and document with: "Currently the pattern matching is on the Boogie names that are generated by Dafny. If you're not sure how those are generated, please add * and * at the start and end of your pattern" ?

> But it's convenient to be able to match on specific prefixes

The situation in which you want to match everything with a particular prefix, but that prefix is also the substring of a symbol you don't want to match, seems rare to me. I would guess that supporting that is less important than making this feature more user friendly through the automatic pre and post wildcards, but I'm OK with both. Let me know which one you prefer.

Reply:
I prefer the version where we don't introduce the wildcards; it's relatively common to have methods an modules named the same, which causes collisions in `/proc`

Reply:
> it's relatively common to have methods an modules named the same

If there's a method `Y` with fully qualified name `X.Y` and a module `Y`, then there's no problem right? `--filter=X.Y` still works. What's the common situation you're worried about?

Reply:
As per the above I think we should not add the `*`s as long as we're clearer we're filtering to Boogie names.